src/views/virtualmachines/actions/components/CloneVMModal/CloneVMModal.tsx

```diff
@@ -87,21 +87,29 @@ const CloneVMModal: React.FC<CloneVMModalProps> = ({ vm, isOpen, onClose }) => {
       await stopVM(vm);
     }
 
-    const cloneRevisionInstanceType = await cloneControllerRevision(
-      updatedVM?.spec?.instancetype?.revisionName,
-      updatedVM.metadata.namespace,
-      vm.metadata.namespace,
-    );
-    const cloneRevisionPreference = await cloneControllerRevision(
-      updatedVM?.spec?.preference?.revisionName,
-      updatedVM.metadata.namespace,
-      vm.metadata.namespace,
-    );
+    const [cloneRevisionInstanceType, cloneRevisionPreference] = await Promise.all([
+      cloneControllerRevision(
+        updatedVM?.spec?.instancetype?.revisionName,
+        updatedVM.metadata.namespace,
+        vm.metadata.namespace,
+      ),
+      cloneControllerRevision(
+        updatedVM?.spec?.preference?.revisionName,
+        updatedVM.metadata.namespace,
+        vm.metadata.namespace,
+      ),
+    ]);
 
     const createdVM = await k8sCreate({ model: VirtualMachineModel, data: updatedVM });
 
-    await updateControllerRevisionOwnerReference(cloneRevisionInstanceType, createdVM);
-    await updateControllerRevisionOwnerReference(cloneRevisionPreference, createdVM);
+    await Promise.all([
+      cloneRevisionInstanceType
+        ? updateControllerRevisionOwnerReference(cloneRevisionInstanceType, createdVM)
```

Comment:
Maybe in updateControllerRevisionOwnerReference check the first arg exists if not return resolve?

Reply:
fixed